compiler/src/iree/compiler/Codegen/Common/RematerializeParallelOps.cpp

```diff
@@ -25,6 +25,34 @@ static bool isScalarOrTensorOfSizeOne(Type t) {
   return t.isIntOrIndexOrFloat();
 }
 
+static bool hasExternalCapture(linalg::GenericOp genericOp) {
```

Comment:
Could you add a comment explaining what this does and maybe provide an example of a generic with an external capture?